### DISCUSSION TITLE
Adding rows in angular data grid, how to choose what column gets focused  when entering edit mode?

### DISCUSSION THREAD

**Discussion by Ugla-Bugla** [IgniteUI/igniteui-angular#12500] [Q&A] [answered]
Is it possible to control which column that gets focused when entering edit mode after pressing add row button?

ex:
in the example below, product id is skipped and editing starts in reorder level. I would like it to further skip reorder level and start in produt name so that if i start typing i will be typing out the product name right away.

![image](https://user-images.githubusercontent.com/63595471/210997336-26c043cb-edb3-4458-b2c0-a507504bef47.png)
https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/row-adding

**Comment by georgianastasov** [comment, marked as answer]
Hi @Ugla-Bugla, 

I have been looking into your question and in order to achieve these requirements, what I suggest as an approach is to make all columns before the one from which you want to start editing and adding a row non-editable with [editable](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxColumnComponent.html#editable) property of igx-column component set to false.
```
<igx-column
      field="ReorderLevel"
      header="ReorderLever"
      dataType="number"
      [editable]="false"
></igx-column>
```

Thus, when adding or editing a row, it will start from the first column that is editable.

After that you will handle the [rowEditEnter](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#rowEditEnter) and [rowEditExit](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#rowEditExit) events. When editing or adding a row and when the rowEditEnter event is fired, you will set those columns that are non-editable back to editable so that you can add or edit values in these cells.

```
public editStart(event){
        this.grid.getColumnByName("ReorderLevel").editable = true;
}
```

And when exiting from editing or adding a row, when rowEditExit event is fired, you will again set the same columns to non-editable so that when these events are performed next time, they can start again from the column that is according to your requirements.

```
public editExit(event){
        this.grid.getColumnByName("ReorderLevel").editable = false;
}
```

In addition, I have prepared small sample illustrating this behavior which could be found [here](https://stackblitz.com/edit/angular-bzx8yk?file=src/app/grid/grid-add-row-sample/grid-add-row-sample.component.html). Please test it on your side and let me know how it behaves.

The described scenario could be observed here:


![edit-add](https://user-images.githubusercontent.com/48180072/211544002-af9d695b-8199-4a8f-995e-60578d99cf0e.gif)

**Reply by Ugla-Bugla** [reply]
Thanks for the thorough reply.

**Comment by Eralmidia** [comment]
While this do work, it feels kinda like a hack tbh. It would be great to have an input on the grid with the name of the field which should get focus when edit starts.

**Reply by georgianastasov** [reply]
Hello @Eralmidia,

After investigating this matter further and doing some research, the input property for the name of the field which should get focus on edit start has been determined to be a feature request. You could submit your idea for feature request in our GitHub repository [here](https://github.com/IgniteUI/igniteui-angular/issues/new?assignees=&labels=feature-request&template=feature_request.md&title=).

Remember when submitting your idea to explain the context in which a feature would be used and why it is needed as well as anything that would prevent you from accomplishing this today.

This will give you the opportunity to directly communicate with our development team regarding the feature request and get notifications whenever new information is available.
